coding-challenges.js

```diff
@@ -2,59 +2,162 @@
+    // stores a random number between 0 and the variable i as declared above
+    const j = Math.floor(Math.random() * i);
+    // temp variable stores the value of the current element's value
+    const temp = arr[i];
+    // changes/switches the value of the current element
+    // to the value of the random number generated above
+    arr[i] = arr[j];
+    // changes the value of arr[j] to temp's value
+    arr[j] = temp;
+  }
+
+  return arr;
+}
+
+console.log(removeAndShuffleV2(collections))
 // Expected output example1 (can be a different order): ["yellow", "blue", "pink", "green"]
 // Expected output example2 (can be a different order): ["blue", "green", "pink"]
 // Expected output example3 (can be a different order): ["pink", "green"]
 
 
-// --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
-
-
 
 
 
+// --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 var cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
 var cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
 
 
+const sumCubed = arr => {
+  // (2 * 2 * 2) + (3 * 3 * 3) + (4 * 4 * 4) = 99
+  // 8 + 27 + 64 = 99
+  // maps thru and cubed each element
+  // used reduce to accumulate elements
+  // must set intial value to 0 or it will run incorrectly
+  return arr.reduce((acc, val) => acc + (val ** 3), 0);
+  // initial value = 0
+  // intiVal + (2**3) = 8
+  // 8 becomes the accumulator (acc)
+  // acc + (3**3) = 35
+  // 35 becomes the acc
+  // acc + (4**3) = 99
+}
+
+console.log(sumCubed(cubeAndSum1));
+console.log(sumCubed(cubeAndSum2));
 
 // --------------------3) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
-
-
-
-
-
 var nums1 = [3, 56, 90, -8, 0, 23, 6]
 // Expected output: [-8, 90]
 var nums2 = [109, 5, 9, -59, 8, 24]
 // Expected output: [-59, 109]
 
+const minMax = arr => {
+  return [Math.min(...arr), Math.max(...arr)];
+}
 
-// --------------------4) Create a function that takes in a string and returns a string with every other letter capitalized.
-
-
+console.log(minMax(nums1));
+console.log(minMax(nums2));
 
 
 
+// --------------------4) Create a function that takes in a string and returns a string with every other letter capitalized.
 var testString1 = "albatross"
 // Expected output: "aLbAtRoSs"
 var testString2 = "jabberwocky"
 // Expected output: "jAbBeRwOcKy"
 
 
-// --------------------5) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator.
+// VERSION 1.0
+const coolLettersV1 = str => {
+  // converts string to array
+  let newStr = str.split("");
+  // loops through array toUpperCase in odd indexes
+  for (let i = 1; i < newStr.length; i += 2) {
+    newStr[i] = newStr[i].toUpperCase();
+  }
+  // join array back into a single string
+  return newStr.join("");
```

Comment:
cool approach! i haven't seen this one yet. remember you don't have to split a string to iterate through it though :) you can just loop through a string like normal